pkg/operation/botanist/addons/nginxingress/nginxingress.go

```diff
@@ -0,0 +1,699 @@
+				Namespace: n.namespace,
+				Labels: map[string]string{
+					v1beta1constants.LabelApp: labelAppValue,
+					labelKeyRelease:           labelValueAddons,
+				},
+				Annotations: map[string]string{
+					resourcesv1alpha1.DeleteOnInvalidUpdate: "true",
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "Role",
+				Name:     role.Name,
+			},
+			Subjects: []rbacv1.Subject{{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      serviceAccount.Name,
+				Namespace: serviceAccount.Namespace,
+			}},
+		}
+
+		serviceController = &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      serviceNameController,
+				Namespace: n.namespace,
+				Labels: map[string]string{
+					v1beta1constants.LabelApp: labelAppValue,
+					labelKeyRelease:           labelValueAddons,
+					labelKeyComponent:         labelValueController,
+				},
+				Annotations: map[string]string{"service.beta.kubernetes.io/aws-load-balancer-proxy-protocol": "*"},
+			},
+			Spec: corev1.ServiceSpec{
+				Type:                     corev1.ServiceTypeLoadBalancer,
+				LoadBalancerSourceRanges: n.values.LoadBalancerSourceRanges,
+				ExternalTrafficPolicy:    n.values.ExternalTrafficPolicy,
+				Ports: []corev1.ServicePort{
+					{
+						Name:       "http",
+						Port:       servicePortControllerHttp,
+						Protocol:   corev1.ProtocolTCP,
+						TargetPort: intstr.FromInt(int(containerPortControllerHttp)),
+					},
+					{
+						Name:       "https",
+						Port:       servicePortControllerHttps,
+						Protocol:   corev1.ProtocolTCP,
+						TargetPort: intstr.FromInt(int(containerPortControllerHttps)),
+					},
+				},
+				Selector: map[string]string{
+					v1beta1constants.LabelApp: labelAppValue,
+					labelKeyRelease:           labelValueAddons,
+					labelKeyComponent:         labelValueController,
+				},
+			},
+		}
+
+		serviceBackend = &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: serviceNameBackend,
+				Labels: map[string]string{
+					v1beta1constants.LabelApp: labelAppValue,
+					labelKeyComponent:         labelValueBackend,
+					labelKeyRelease:           labelValueAddons,
+				},
+				Namespace: n.namespace,
+			},
+			Spec: corev1.ServiceSpec{
+				Type: corev1.ServiceTypeClusterIP,
+				Ports: []corev1.ServicePort{{
+					Port:       servicePortBackend,
+					TargetPort: intstr.FromInt(int(containerPortBackend)),
+				}},
+				Selector: map[string]string{
+					v1beta1constants.LabelApp: labelAppValue,
+					labelKeyComponent:         labelValueBackend,
+					labelKeyRelease:           labelValueAddons,
+				},
+			},
+		}
+
+		ingressClass   *networkingv1.IngressClass
+		vpa            *vpaautoscalingv1.VerticalPodAutoscaler
+		roleBindingPSP *rbacv1.RoleBinding
+	)
+
+	if n.values.VPAEnabled {
+		updateMode := vpaautoscalingv1.UpdateModeAuto
+		vpa = &vpaautoscalingv1.VerticalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      vpaName,
+				Namespace: n.namespace,
```

Comment:
```suggestion
				Namespace: addonNamespace,
```
nginx-ingress-addon should be deployed in `kube-system` namespace of the shoot.